testing/tests/cpp_nesting.py

```diff
@@ -119,10 +119,42 @@ def check_most_pages(self, uses_mainpage=True, main_refid="indexpage"):
         # TODO: make this a parameter if we add more than two tests...
         if uses_mainpage:
             f_name = "page_town_rock.hpp"
+            # Not sure why, but when it's a mainpage it's called "index" and when it's not it's called "overview"
+            # this is a bit hacky but seems to fix it.
+            custom_page = "index"
         else:
             f_name = "page_town_rock_alt.hpp"
-        file_hierarchy_dict[include_dir][file(f_name)] = {}
-        compare_file_hierarchy(self, file_hierarchy(file_hierarchy_dict))
+            custom_page = "overview"
+        file_hierarchy_dict[include_dir][file(f_name)] = {
+            page(custom_page): {
+                page("intro"): {
+                    page("more_nesting"): {},
+                    page("more_nesting_redux"): {
+                        page("more_nesting_redux_again"): {},
+                        page("more_nesting_redux_again_again"): {},
+                    },
+                },
+                page("advanced"): {}
+            },
+        }
+        file_hierarchy_temp = file_hierarchy(file_hierarchy_dict)
+
+        def flattenHierarchyFile(file):
```

Comment:
So the tests were failing because exhale adds every page/child to the first level of the listing in a file group. I first tried adding all the nested children to the top-level, but that then caused the tests to fail since the children now had no parents.
This function takes the properly nested/initialized hierarchy above and shallow-copies the children of each node to the topmost level so that you get every single node in the page at the top level.